Neos.Neos/Classes/Routing/FrontendNodeRoutePartHandler.php

```diff
@@ -225,18 +225,19 @@ protected function convertRequestPathToNode($requestPath)
      * absolute node path: /sites/neostypo3org/homepage/about@user-admin
      * $this->value:       homepage/about@user-admin
      *
-     * @param mixed $node Either a Node object or an absolute context node path
+     * @param NodeInterface|string[] $node Either a Node object or an absolute context node path wrapped in an array
      * @return bool|ResolveResult An instance of ResolveResult if the route coulr resolve the $node, otherwise FALSE. @see DynamicRoutePart::resolveValue()
      * @throws Exception\NoSiteException | InvalidRequestPathException | NeosException | IllegalObjectTypeException
+     * @see NodeIdentityConverterAspect
      */
     protected function resolveValue($node)
     {
-        if (!$node instanceof NodeInterface && !is_string($node)) {
+        if (!$node instanceof NodeInterface && !is_array($node)) {
             return false;
         }
 
-        if (is_string($node)) {
-            $nodeContextPath = $node;
+        if (is_array($node) && isset($node['__contextNodePath'])) {
+            $nodeContextPath = $node['__contextNodePath'];
```

Comment:
I don't really understand that change to be honest. The `FrontendRoutePartHandler` should only be used for nodes of a public workspace (`live` usually). For anything else the `preview` route is to be used.

We have to support values of type `string` for the resolve case so that you can do sth like:
```yaml
-
  name: 'My custom route'
  uriPattern: '{node}/foo'
  defaults:
    node: '/sites/some-site/some/node'
  routeParts:
    'node':
      handler: 'Neos\Neos\Routing\FrontendNodeRoutePartHandlerInterface'
```

But why do we need to support arrays?

Reply:
> We have to support values of type string for the resolve case so that you can do sth like:

TBH I never saw such a setup. Is that really something we _have_ to support? 🤔

Reply:
I've seen it in some documentation about "beautiful plugin routes"

Reply:
With your commit that should be fine - I like it.